infrastructure/infrastructure/src/main/java/alabno/wserver/WebSocketHandler.java

```diff
@@ -75,10 +75,32 @@ public void handleMessage(WebSocket conn, String message) {
     }
 
     private void handleFeedback(JsonParser parser, WebSocket conn) {
-        String source = parser.getString("source");
-        String annType = parser.getString("ann_type");
-        String annotation = parser.getString("annotation");
-        this.updaters.updateAll(source, annType, annotation);
+        try {
+            String source = parser.getString("source");
+            String annType = parser.getString("ann_type");
+            String annotation = parser.getString("annotation");
+
+            String fileName = parser.getString("filename");
+            int lineno = parser.getInt("lineno");
+            String token = parser.getString("id");
+            
+            amendFile(fileName, lineno, annType, annotation, token);
+            
+            updaters.updateAll(source, annType, annotation);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void amendFile(String fileName, int lineno, String annType, String annotation, String token) {
```

Comment:
I think you should have an annotation class with a JSON representation so that you don't have to do this in 128293 files